### PR TITLE
feat: integrate forc-wallet and forc-deploy

### LINKED ISSUE #4743: forc-deploy should be integrated with forc-wallet
Current flow is bad in terms of UX as users require two different terminal sessions and copy paste signature artifacts between two terminals.

#4723 have pretty good details.

### PR DESCRIPTION
**Pull request by kayagokalp**
## Description
closes #4743.

This PR integrates forc-wallet and forc-deploy together so that forc-deploy can offer to sign your transactions without running a second terminal window. Below there are some inputs and outputs after this change.

```console
forc-deploy --random-salt
```

```console
Contract id: 0x77e0c380dff45759d1a16b6add83d0fd38e8d86ddd0c96cd5117c82b6d958edb

Please provide the password of your encrypted wallet vault at "/Users/kayagokalp/.fuel/wallets/.wallet":
```

At this point if the user enters wrong password:
```console
Contract id: 0x77e0c380dff45759d1a16b6add83d0fd38e8d86ddd0c96cd5117c82b6d958edb

Please provide the password of your encrypted wallet vault at "/Users/kayagokalp/.fuel/wallets/.wallet":
Error: Failed to access forc-wallet vault. Please check your password
```

If the user enters correct password:

```console
Contract id: 0x77e0c380dff45759d1a16b6add83d0fd38e8d86ddd0c96cd5117c82b6d958edb

Please provide the password of your encrypted wallet vault at "/Users/kayagokalp/.fuel/wallets/.wallet":
Do you accept to sign this transaction with fuel15f7mn9vue4eks9znjqf5tg6ylv8vkfh9xcdh3v6z9s7yagewgwasaxpxem [y/N]:
```

If the user enters `y` or `Y` and hits enter:

```console
Contract id: 0x77e0c380dff45759d1a16b6add83d0fd38e8d86ddd0c96cd5117c82b6d958edb

Please provide the password of your encrypted wallet vault at "/Users/kayagokalp/.fuel/wallets/.wallet":
Do you accept to sign this transaction with fuel15f7mn9vue4eks9znjqf5tg6ylv8vkfh9xcdh3v6z9s7yagewgwasaxpxem [y/N]:y
contract 98f158f121c10bdfea5ffbdda904d3e7b61c7f9d80ae2beff75b5d04f0f3547b deployed in block 0x1dd25813f6a82a69347079feb265eeeb37e89f992cea65ad1373b378b80c1ecb
```
If the user enters anything else:

```console
Contract id: 0x77e0c380dff45759d1a16b6add83d0fd38e8d86ddd0c96cd5117c82b6d958edb

Please provide the password of your encrypted wallet vault at "/Users/kayagokalp/.fuel/wallets/.wallet":
Do you accept to sign this transaction with fuel15f7mn9vue4eks9znjqf5tg6ylv8vkfh9xcdh3v6z9s7yagewgwasaxpxem [y/N]:N
Error: User refused to sign
```

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: forc-plugins/forc-client/src/util/tx.rs
Comment (sdankel) [inline forc-plugins/forc-client/src/util/tx.rs:149]: nit: if we have issues for any of these TODOs, we should reference the issue number in the comment.

**Review by sdankel — DISMISSED** [review]
utACK

File: forc-plugins/forc-client/src/util/tx.rs
Comment (kayagokalp) [reply forc-plugins/forc-client/src/util/tx.rs:149]: We do not have them currently. I will be opening them tomorrow morning and will be tackling these todos as the first thing in the next PR

**Comment by kayagokalp** [comment]
Also will keep this waiting, won't merge right away because I need a release for fuel.nix stuff with #4750 merged. I want to omit releasing a version with this (before updating the UX). So my current plan looks like:

1. Merge #4750
2. Cut a sway repo release
3. Merge this one
4. Merge UX improvements
5. Cut another release

So that if anything comes up (very bad bugs with this etc) we do not block fuel.nix resolutions any longer, and have a release  that both works for fuel.nix and without this changes.

File: forc-plugins/forc-client/src/util/tx.rs
Comment (JoshuaBatty) [inline forc-plugins/forc-client/src/util/tx.rs:143]: Perhaps here instead of bailing, we could prompt the user is they would like a new wallet generated for them. If they select no then we can return at that point.

Comment (kayagokalp) [reply forc-plugins/forc-client/src/util/tx.rs:143]: It sounds good! but I have to expose that part as well. For the improved UX PR that I will have next, I have couple of other things to expose as well. Such as listing accounts so that users can select which account to sign with and balance queries so that we can offer them to go to faucet. I can add expose creating a new wallet alongside them or if you think we should have it with this PR, we will have to wait for forc-wallet changes first.

Comment (JoshuaBatty) [reply forc-plugins/forc-client/src/util/tx.rs:143]: That's fine we can do that in a follow up PR. Maybe just open an issue so we don't forget.

**Review by JoshuaBatty — DISMISSED** [review]

**Comment by JoshuaBatty** [comment]
Instead of typing `forc-deploy --random-salt` can `--random-salt` just be on by default? Perhaps this is already the case was just confused that it was in the description.

**Comment by kayagokalp** [comment]
Opened more granular issues such as:

1. #4759 
2. #4758 
3. #4757 
4. #4756 
5. #4754

**Review by JoshuaBatty — APPROVED** [review]

**Review by vaivaswatha — APPROVED** [review]
Approving just for admin reasons (two reviews needed) as this was already approved before and just had a merge conflict fixed.

**Pull request merged**
